File: Server/src/main/content/global/skill/gathering/hunter/HunterGear.java
package content.global.skill.gathering.hunter;

import core.game.node.entity.player.Player;
import core.game.node.item.Item;

/**
 * The enum Hunter gear.
 */
public enum HunterGear {
    /**
     * The Glove of silence.
     */
    GLOVE_OF_SILENCE(3.0, new Item(10075)),
    /**
     * The Spotier cape.
     */
    SPOTIER_CAPE(5.0, new Item(10071)),
    /**
     * The Spotted cape.
     */
    SPOTTED_CAPE(5.0, new Item(10069)),
    /**
     * The Larupia.
     */
    LARUPIA(10.00, new Item(10045), new Item(10043), new Item(10041)),
    /**
     * The Desert gear.
     */
    DESERT_GEAR(10.00, new Item(12568), new Item(10063), new Item(10061)),
    /**
     * The Graahk gear.
     */
    GRAAHK_GEAR(10.00, new Item(10051), new Item(10047), new Item(10049)),
    /**
     * The Kyatt gear.
     */
    KYATT_GEAR(10.00, new Item(10039), new Item(10035), new Item(10037)),
    /**
     * The Jungle gear.
     */
    JUNGLE_GEAR(8.00, new Item(10059), new Item(10057)),
    /**
     * The Polar gear.
     */
    POLAR_GEAR(8.00, new Item(10065), new Item(10067));


    HunterGear(double chanceRate, Item... equipment) {
        this.equipment = equipment;
        this.chanceRate = chanceRate;
    }

    private final Item[] equipment;

    private final double chanceRate;

    /**
     * In gear hunter gear.
     *
     * @param player the player
     * @return the hunter gear
     */
    public static HunterGear inGear(final Player player) {
        int contained = 0;
        for (HunterGear type : values()) {
            for (Item i : type.getEquipment()) {
                if (player.getEquipment().containsItem(i)) {
                    contained += 1;
                }
                if (contained == type.getEquipment().length) {
                    return type;
                }
            }
        }
        return null;
    }

    /**
     * Gets chance rate.
     *
     * @param player the player
     * @return the chance rate
     */
    public static double getChanceRate(Player player) {
        final HunterGear gear = inGear(player);
        if (gear == null) {
            return 0.0;
        }
        return gear.getChanceRate();
    }

    /**
     * Get equipment item [ ].
     *
     * @return the item [ ]
     */
    public Item[] getEquipment() {
        return equipment;
    }

    /**
     * Gets chance rate.
     *
     * @return the chance rate
     */
    public double getChanceRate() {
        return chanceRate;
    }
}

File: Server/src/main/content/global/skill/gathering/hunter/HunterGear.kt
package content.global.skill.gathering.hunter

import core.game.node.entity.player.Player
import core.game.node.item.Item

/**
 * Represents a set of hunter gear.
 * @author Vexia
 *
 *
 * @param chanceRate Represents the probability of a successful hunt with this gear.
 * @constructor Represents the HunterGear enum with a chance rate and associated equipment.
 *
 * @param equipment Vararg parameter that allows multiple items to be associated with the gear.
 */
enum class HunterGear(val chanceRate: Double, vararg equipment: Item) {
    /**
     * Glove Of Silence.
     */
    GLOVE_OF_SILENCE(3.0, Item(10075)),

    /**
     * Spotier Cape.
     */
    SPOTIER_CAPE(5.0, Item(10071)),

    /**
     * Spotted Cape.
     */
    SPOTTED_CAPE(5.0, Item(10069)),

    /**
     * Larupia.
     */
    LARUPIA(10.00, Item(10045), Item(10043), Item(10041)),

    /**
     * Desert Gear.
     */
    DESERT_GEAR(10.00, Item(12568), Item(10063), Item(10061)),

    /**
     * Graahk Gear.
     */
    GRAAHK_GEAR(10.00, Item(10051), Item(10047), Item(10049)),

    /**
     * Kyatt Gear.
     */
    KYATT_GEAR(10.00, Item(10039), Item(10035), Item(10037)),

    /**
     * Jungle Gear.
     */
    JUNGLE_GEAR(8.00, Item(10059), Item(10057)),

    /**
     * Polar Gear.
     */
    POLAR_GEAR(8.00, Item(10065), Item(10067));

    val equipment: Array<Item> = equipment as Array<Item>

    companion object {
        /**
         * Method used to check if the player in the gear.
         * @param player the player.
         * @return the gear.
         */
        @JvmStatic
        fun inGear(player: Player): HunterGear? {
            var contained = 0
            for (type in values()) {
                for (i in type.equipment) {
                    if (player.equipment.containsItem(i)) {
                        contained += 1
                    }
                    if (contained == type.equipment.size) {
                        return type
                    }
                }
            }
            return null
        }
        /**
         * Gets the chance rate the player has.
         * @param player the player.
         * @return the rate.
         */
        @JvmStatic
        fun getChanceRate(player: Player): Double {
            val gear = inGear(player) ?: return 0.0
            return gear.chanceRate
        }
    }
}